### PR TITLE
Turn threaded price source background thread into async task

### PR DESCRIPTION
**Pull request by e00E**
using async-std.
Simplified the threaded price source in several places:
* get rid of code only compiled for tests
* use weak ptr to observe dropping of parent struct

Now that we have async_std I turned futures::Mutex into async_std::Mutex
and futures_timer::sleep into async_std::sleep. If this gets merged we
can do the same changes in other places and remove the futures-timer
dependency.

### Test Plan
CI

### CODE REVIEW

File: core/src/price_estimation/threaded_price_source.rs [hunk 1]
```diff
@@ -1,90 +1,48 @@
...
-                        // Make sure we don't hold the mutex while this blocking call happens.
-                        match price_source.get_prices(&tokens).wait() {
-                            Ok(prices) => price_map.lock().wait().extend(prices),
-                            Err(err) => log::warn!("price_source::get_prices failed: {}", err),
-                        }
+        let join_handle = task::spawn({
+            let price_map = Arc::downgrade(&price_map);
+            async move {
+                while let Some(price_map) = price_map.upgrade() {
+                    match price_source.get_prices(&tokens).wait() {
```
Comment (fleupold) [inline core/src/price_estimation/threaded_price_source.rs:37]: Should this not be an await?

File: core/src/price_estimation/threaded_price_source.rs [hunk 2]
```diff
@@ -1,90 +1,48 @@
...
-            move || loop {
-                match receiver.recv_timeout(update_interval) {
-                    Ok(()) | Err(RecvTimeoutError::Timeout) => {
-                        // Make sure we don't hold the mutex while this blocking call happens.
-                        match price_source.get_prices(&tokens).wait() {
-                            Ok(prices) => price_map.lock().wait().extend(prices),
-                            Err(err) => log::warn!("price_source::get_prices failed: {}", err),
-                        }
+        let join_handle = task::spawn({
+            let price_map = Arc::downgrade(&price_map);
```
Comment (fleupold) [inline core/src/price_estimation/threaded_price_source.rs:34]: 👍Elegant!

File: core/src/price_estimation/threaded_price_source.rs [hunk 3]
```diff
@@ -1,90 +1,48 @@
...
+            async move {
+                while let Some(price_map) = price_map.upgrade() {
+                    match price_source.get_prices(&tokens).wait() {
+                        Ok(prices) => price_map.lock().wait().extend(prices),
```
Comment (fleupold) [inline core/src/price_estimation/threaded_price_source.rs:38]: Does MutexGuard automatically convert to a Promis or what 🧙is happening here? Can this also be an await?

**Review by fleupold — APPROVED** [review]

File: core/src/price_estimation/threaded_price_source.rs (hunk 1 above)
Comment (e00E) [reply core/src/price_estimation/threaded_price_source.rs:37]: Good thing you saw that. All `wait()`s should be await. This was leftover from the threaded code.

**Review by nlordell — APPROVED** [review]
I really like this change.

**Pull request merged**